dev_docs: Document the process for preparing icons for Zulip use.

karlstolley opened the pull request:
This PR adds some finesse to the icons section of the contributor docs, and also introduces a new, illustrated section outlining the process of preparing an icon for use with Zulip. It covers two common icon problems: icons that make use of strokes, and icons that include a problem `evenodd` value on `fill-rule`.

Care has been taken to avoid being overly descriptive of Inkscape's interface, so that this documentation might survive future changes to Inkscape.

[CZO discussion](https://chat.zulip.org/#narrow/channel/6-frontend/topic/SVG.20for.20.2332335/near/2158685)

![docs-preparing-icons-for-zulip](https://github.com/user-attachments/assets/d8020950-0bc6-40f0-88a4-65c2e7cc7830)

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

alya commented:
Looks good to me, so can be marked for integration review once @andersk approves.

andersk commented on docs/subsystems/icons.md:
```diff
@@ -30,28 +30,137 @@ using icons. Typically, this means:
+
+1. Strip back the opening of the file to include only the `<svg>` tag and its `xmlns`,
+   `width`, `height`, `viewBox`, and `fill` attributes and their values as set by
+   Inkscape. The one exception there is the `fill` attribute's value, which you should
+   set to `#000000`: while that has no bearing on how the icon will render in Zulip, it
+   makes it easy to see the icon in operating-system previews and elsewhere.
```
`fill` is unnecessary. The [initial value](https://www.w3.org/TR/SVG2/painting.html#SpecifyingFillPaint) is already black.

andersk commented:
Another issue we sometimes run into is `fill-rule: evenodd`, e.g. [here](https://chat.zulip.org/#narrow/channel/101-design/topic/UI.20redesign.3A.20compose.20icons/near/1689053).

karlstolley commented:
Great, thank you for the feedback, @andersk. I've reworked this to include a section on cleaning up the `evenodd` fill rule, and moved the file-cleanup instructions to its own section.

@alya, given the restructuring and additional content here, I'd appreciate another read-through from you when you have the time.

alya commented on docs/subsystems/icons.md:
```diff
@@ -30,28 +30,186 @@ using icons. Typically, this means:
+1. Open the SVG file in [Inkscape](https://inkscape.org/), a free and open-source
+   visual editor for vector images, including SVG.
+
+1. Use Ctrl + A or otherwise select all components of the icon.
```
Style the keyboard shortcut.

alya commented on docs/subsystems/icons.md:
```diff
@@ -30,28 +30,186 @@ using icons. Typically, this means:
+   on the `fill-rule` attribute
+
+Each of those problems requires a different fix.
+
```
I think it should be mentioned here that you'll need to clean up the SVG after either fix.

karlstolley commented:
Thanks for the feedback, @alya! I've addressed each of your points (capitalized first letters on list items; backticks around keyboard shortcuts; mentioning the need for manual cleanup, with a link to that section) in the latest push.

timabbott commented:
Merged, thanks @karlstolley and @andersk!
